src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,150 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
     return true;
 }
 
-bool emberAfGroupsClusterGetGroupMembershipCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
-                                                    const Commands::GetGroupMembership::DecodableType & commandData)
+struct GetGroupMembershipResponse
 {
-    auto & groupCount = commandData.groupCount;
-    auto & groupList  = commandData.groupList;
+public:
+    // Use GetCommandId instead of commandId directly to avoid naming conflict with CommandIdentification in ExecutionOfACommand
+    static constexpr CommandId GetCommandId() { return Commands::GetGroupMembershipResponse::Id; }
+    static constexpr ClusterId GetClusterId() { return Groups::Id; }
 
-    EmberStatus status = EMBER_ZCL_STATUS_FAILURE;
-    uint8_t i, j;
-    uint8_t count = 0;
-    uint8_t list[EMBER_BINDING_TABLE_SIZE << 1];
-    uint8_t listLen = 0;
-    CHIP_ERROR err  = CHIP_NO_ERROR;
+    GetGroupMembershipResponse(const Commands::GetGroupMembership::DecodableType & data,
+                               GroupDataProvider::GroupMappingIterator * iter) :
+        mCommandData(data),
+        mIterator(iter)
+    {}
 
-    emberAfGroupsClusterPrint("RX: GetGroupMembership [");
+    const Commands::GetGroupMembership::DecodableType & mCommandData;
+    GroupDataProvider::GroupMappingIterator * mIterator = nullptr;
 
-    // For all networks, Get Group Membership commands may be sent either
-    // unicast or broadcast (removing the ZLL-specific limitation to unicast).
+    uint8_t mCapacity = 0xff;
```

Comment:
Well, OK, now we are using `mCapacity`, but the comment is not in the right place... Basically, the comment needs to be next to the place where we decide to send `0xff`.  If we are encoding `mCapacity`, the comment should be on `mCapacity`.  Back when we were encoding a literal `0xff`, the comment needed to be next to that literal.

Reply:
I have move the comment right before the Encode call for mCapacity.